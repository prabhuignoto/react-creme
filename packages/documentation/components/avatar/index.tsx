import { faUser } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import DemoPageRenderer from '../../common/demo-page-renderer';
import { Widgets } from './widgets';

const Description = (
  <div>
    <p className="rc-component-description">
      {`Avatar is a component that displays a user's profile picture or initials.`}
    </p>
  </div>
);

const Accordion = () => {
  return (
    <DemoPageRenderer
      description={Description}
      title="Avatar"
      tabTitles={['Examples', 'Properties', 'Stackblitz']}
      stackBlitzCodes={['react-ts-s9qxmk']}
      sourceId="avatar/avatar.tsx"
      editId="avatar"
      features={[
        'Custom Avatar sizes: small, medium and large',
        'Support for rendering custom icon',
      ]}
      pageIcon={<FontAwesomeIcon icon={faUser} size="2x" />}
      properties={[
        {
          default: 'auto generated',
          description: 'Unique id for the accordion',
          name: 'id',
          optional: 'Yes',
          type: 'String',
        },
        {
          default: 'false',
          description: 'Aligns the expand/collapse icon to the Right',
          name: 'alignIconRight',
          optional: 'Yes',
          type: 'Boolean',
        },
        {
          default: "''",
          description: 'Title of the accordion',
          name: 'title',
          optional: 'Yes',
          type: 'String',
        },
        {
          default: 'chevron',
          description: `Prop to change the icon type. <br> <em>chevron</em> | <em>plus</em>`,
          name: 'iconType',
          optional: 'Yes',
          type: 'String',
        },
        {
          default: 'false',
          description: 'Enable or Disable the border around the accordion',
          name: 'border',
          optional: 'Yes',
          type: 'Boolean',
        },
        {
          default: 'cubic-bezier(0.19, 1, 0.22, 1)',
          description: 'Custom transition function',
          name: 'transition',
          optional: 'Yes',
          type: 'String',
        },
      ]}
      demoWidget={<Widgets />}
    ></DemoPageRenderer>
  );
};

export default Accordion;
