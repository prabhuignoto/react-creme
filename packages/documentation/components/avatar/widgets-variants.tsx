import { RateIcon } from '@icons';
import { Avatar } from '@lib';

export const Default = <Avatar />;

export const MediumSized = <Avatar size="md" />;

export const LargeSized = <Avatar size="lg" />;

export const CustomIcon = (
  <Avatar size="lg">
    <RateIcon />
  </Avatar>
);

export const Letter = <Avatar size="lg" letter="A" />;
