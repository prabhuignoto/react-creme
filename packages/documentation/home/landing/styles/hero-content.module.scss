@use '@design/theme.scss';

.content {
  border-radius: 8px;
  padding: 3rem;
}

@mixin grid {
  column-gap: 2rem;
  display: grid;
  grid-template-columns: 1fr 2fr;
  justify-items: center;
}

@media (min-width: 768px) {
  .content {
    max-width: 600px;
  }
}

@media (min-width: 1024px) {
  .content {
    display: flex;
    flex-direction: column;
    max-width: 800px;
  }
}

@media (min-width: 1280px) {
  .content {
    @include grid;
    max-width: 1000px;
  }
}

@media (min-width: 1440px) {
  .content {
    @include grid;
    max-width: 1240px;
  }
}

@media (min-width: 1920px) {
  .content {
    @include grid;
    max-width: 1700px;
  }
}

@media (min-width: 2560px) {
  .content {
    @include grid;
    max-width: 2000px;
  }
}

.left_column {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  min-width: 400px;
}

.right_column {
  align-items: center;
  align-self: flex-end;
  display: grid;
  gap: 1rem;
  grid-template-columns: 1fr 1fr;
  justify-items: center;
  width: 100%;
}

.header {
  color: theme.$primary;
  font-size: 3rem;
  font-weight: 500;
  margin: 0.2rem 0;
  text-transform: uppercase;
}

.sub_text {
  font-size: 2rem;
  font-weight: 300;
  margin: 1rem 0;

  &.dark {
    color: #fff;
  }
}

.get_started_btn {
  background-color: var(--rc-secondary-color-hex);
  border: 0;
  border-radius: 4px;
  color: var(--rc-color-white);
  font-size: 1.2rem;
  margin-top: 3rem;
  padding: 1rem;
  width: 200px;
}

.buttons {
  display: flex;
  margin: 3rem 0 0;

  > :first-child {
    margin-right: 1rem;
  }
}
