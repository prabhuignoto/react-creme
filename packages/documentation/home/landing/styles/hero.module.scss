.hero {
  align-items: flex-start;
  display: flex;
  justify-content: center;
  width: 100%;

  @media (min-width: 1rem) {
    min-height: 1200px;
  }

  @media (min-width: 43rem) {
    min-height: 980px;
  }

  @media (min-width: 768px) {
    min-height: 1080px;
  }

  @media (min-width: 1280px) {
    min-height: 820px;
  }

  &:not(.dark) {
    background: linear-gradient(
      to left bottom,
      rgba(#bfdbf7, 1) 20%,
      rgba(#e1e5f2, 1)
    );
  }

  &.dark {
    background: linear-gradient(
      to right bottom,
      rgba(#b8c0ff, 0.3) 10%,
      rgba(#1e1e1e, 0.9)
    );
  }
}
