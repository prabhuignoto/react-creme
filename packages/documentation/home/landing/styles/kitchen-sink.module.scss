.sink {
  align-items: center;
  // box-shadow: 0px 0px 10px 5px rgba(#000, 0.1);
  background: #fff;
  border-radius: 4px;
  display: grid;
  height: 300px;
  justify-items: center;
  justify-items: center;
  padding: 1rem;
  width: calc(100% - 2rem);

  &.dark {
    background: #1e1e1e;
  }
}

.sink1 {
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: 1fr 1fr;
}

.sink2 {
  grid-template-columns: 1fr;
}

.sink3 {
  grid-column: 1 / 3;
  grid-template-columns: 1fr 1fr;
  height: 350px;
}

.widget {
  align-items: center;
  border-radius: 4px;
  // padding: 1rem;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  width: 90%;
}

.widget_wrap {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  width: 100%;
}

.widget_wrap_2 {
  align-items: center;
  display: flex;
  justify-content: space-evenly;
  width: 100%;
}
