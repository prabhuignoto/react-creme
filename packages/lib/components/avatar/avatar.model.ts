export type AvatarProps = {
  size?: 'sm' | 'md' | 'lg';
};
