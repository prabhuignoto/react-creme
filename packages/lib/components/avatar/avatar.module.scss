@use '@design/core.scss';
@use '@design/theme.scss';

$size: (sm, md, lg);

.avatar {
  @extend %center;
  background: theme.$primary;
  border-radius: 50%;

  @each $size in $size {
    &.#{$size} {
      height: var(--rc-icon-size-#{$size});
      width: var(--rc-icon-size-#{$size});
    }
  }
}

.icon_container {
  @extend %center;
  color: theme.$white;
  height: 70%;
  width: 70%;

  svg {
    fill: theme.$white;
  }
}
