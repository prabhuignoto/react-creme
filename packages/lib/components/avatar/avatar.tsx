import { FunctionComponent } from 'react';

const Avatar: FunctionComponent = () => {
  return <div></div>;
};

export { Avatar };
