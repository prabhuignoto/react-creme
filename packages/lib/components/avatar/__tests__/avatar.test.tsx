import { render } from '@testing-library/react';
import { Avatar } from '../avatar';
import styles from '../avatar.module.scss';

describe('Avatar', () => {
  it('should render avatar with default props', () => {
    const { container, getByRole } = render(<Avatar />);
    expect(container).toMatchSnapshot();
    expect(getByRole('img')).toBeInTheDocument();
  });

  it('should render custom icon', () => {
    const { getByText } = render(<Avatar>icon</Avatar>);
    expect(getByText('icon')).toBeInTheDocument();
  });

  it('should render medium sized avatar', () => {
    const { getByRole } = render(<Avatar size="md" />);
    expect(getByRole('img')).toHaveClass(styles.md);
  });

  it('should render large sized avatar', () => {
    const { getByRole } = render(<Avatar size="lg" />);
    expect(getByRole('img')).toHaveClass(styles.lg);
  });

  it('should render avatar with letter', () => {
    const { getByText } = render(<Avatar letter="A" />);
    expect(getByText('A')).toBeInTheDocument();
  });
});
