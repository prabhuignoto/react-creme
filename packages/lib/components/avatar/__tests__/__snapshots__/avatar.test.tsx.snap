// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > should render avatar with default props 1`] = `
<div>
  <div
    class="_avatar_61bc79 _sm_61bc79"
    role="img"
  >
    <span
      class="_icon_container_61bc79"
    >
      <svg
        class="feather feather-user"
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
        />
        <circle
          cx="12"
          cy="7"
          r="4"
        />
      </svg>
    </span>
  </div>
</div>
`;
